packages/nuxt/src/core/app.ts

```diff
@@ -111,8 +111,7 @@ export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
       ...(config.plugins || []),
       ...config.srcDir
         ? await resolveFiles(config.srcDir, [
-          `${config.dir?.plugins || 'plugins'}/*.{ts,js,mjs,cjs,mts,cts}`,
-          `${config.dir?.plugins || 'plugins'}/*/index.*{ts,js,mjs,cjs,mts,cts}`
+          `${config.dir?.plugins || 'plugins'}/*.{ts,js,mjs,cjs,mts,cts}`
```

Comment:
I appreciate this PR and the desire to add consistency here. But this would be a breaking change, and this early in Nuxt 3 release cycle it will take some time before we're ready to do that. We could possibly update the PR to show a deprecation warning if anyone is doing this, or even just add a 'TODO' comment to remind us to remove this option in future.

Reply:
I do agree and had the same concern - hence your feedback is important! 🙂

Option B would be to go for the `{composables,utils}/*/index` route (https://github.com/nuxt/framework/pull/10084), but after seeing the previous context to that (https://github.com/nuxt/framework/pull/6025, and antfu's valid point - https://github.com/unjs/unimport/pull/86#issuecomment-1178724831), I'd go for the current approach (even though I'd like my index files to be scanned by default).

Still, both would have side effects - and I'm all for planning removal in a minor/major release after a release with a deprecation warning (along with a TODO comment), but I'm trying to brainstorm how the warning would be generated (figuring out if a file is scanned from `plugins/*/index.*` and it is not explicitly in `nuxt.config` - I may find a solution in a while).